vms/platformvm/state/stakers_ops_test.go

```diff
@@ -0,0 +1,506 @@
+			}
+
+			// store validator
+			store.PutCurrentValidator(&val)
+			retrievedValidator, err := store.GetCurrentValidator(val.SubnetID, val.NodeID) // check version 1
+			if err != nil {
+				return fmt.Sprintf("expected no error, got %v", err)
+			}
+			if !reflect.DeepEqual(&val, retrievedValidator) {
+				return fmt.Sprintf("wrong staker retrieved expected %v, got %v", &val, retrievedValidator)
+			}
+
+			valIt, err := store.GetCurrentStakerIterator() // check version 2
+			if err != nil {
+				return fmt.Sprintf("unexpected failure in staker iterator creation, error %v", err)
+			}
+			if !valIt.Next() {
+				return errNonEmptyIteratorExpected.Error()
+			}
+			if !reflect.DeepEqual(valIt.Value(), retrievedValidator) {
+				return fmt.Sprintf("wrong staker retrieved expected %v, got %v", &val, retrievedValidator)
+			}
+			valIt.Release()
+
+			// store delegators
+			for _, del := range dels {
+				cpy := del
+
+				// it's fine deleting unknown delegator
+				store.DeleteCurrentDelegator(&cpy)
+
+				// finally store the delegator
+				store.PutCurrentDelegator(&cpy)
+			}
+
+			// check no missing delegators by subnetID, nodeID
+			for _, del := range dels {
+				found := false
+				delIt, err := store.GetCurrentDelegatorIterator(subnetID, nodeID)
+				if err != nil {
+					return fmt.Sprintf("unexpected failure in current delegators iterator creation, error %v", err)
+				}
+				for delIt.Next() {
+					if reflect.DeepEqual(*delIt.Value(), del) {
+						found = true
+						break
+					}
+				}
+				delIt.Release()
+
+				if !found {
+					return fmt.Sprintf("missing delegator %v", del)
+				}
+			}
+
+			// check no extra delegator by subnetID, nodeID
+			delIt, err := store.GetCurrentDelegatorIterator(subnetID, nodeID)
+			if err != nil {
+				return fmt.Sprintf("unexpected failure in current delegators iterator creation, error %v", err)
+			}
+			for delIt.Next() {
+				found := false
+				for _, del := range dels {
+					if reflect.DeepEqual(*delIt.Value(), del) {
+						found = true
+						break
+					}
+				}
+				if !found {
+					return fmt.Sprintf("found extra delegator %v", delIt.Value())
+				}
+			}
+			delIt.Release()
+
+			// check no missing delegators in the whole staker set
+			for _, del := range dels {
+				found := false
+				fullDelIt, err := store.GetCurrentStakerIterator()
+				if err != nil {
+					return fmt.Sprintf("unexpected failure in current delegators iterator creation, error %v", err)
+				}
+				for fullDelIt.Next() {
+					if reflect.DeepEqual(*fullDelIt.Value(), del) {
+						found = true
+						break
+					}
+				}
+				fullDelIt.Release()
+
+				if !found {
+					return fmt.Sprintf("missing delegator %v", del)
+				}
+			}
+
+			// delete delegators
+			for _, del := range dels {
+				cpy := del
+				store.DeleteCurrentDelegator(&cpy)
+
+				// check deleted delegator is not there anymore
+				delIt, err := store.GetCurrentDelegatorIterator(subnetID, nodeID)
```

Comment:
would some of the tests in this file be shorter if we had a helper function to iterate all the stakers into an array then check against the expected value?